### PR TITLE
acceptance: support nightlies on Azure

### PR DESCRIPTION
**Pull request by cuongdo**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/12744)

### CODE REVIEW

**Comment by cuongdo** [comment]
I believe I've shaken out most issues and have run the allocator tests successfully. Though I call `ntpdate` before starting `cockroach`, I still occasionally see the following panic:

```
F170106 19:18:42.117697 51 server/server.go:162  [n?] fewer than half the known nodes are within the maximum offset of 250ms (0 of 1)
goroutine 51 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x0, 0x703eb4e, 0x2aff8a0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:849 +0xa5
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2b00ea0, 0xc400000004, 0x22eda4e, 0x10, 0xa2, 0xc4202d09c0, 0x54)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:712 +0x7a8
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7fb3d673e750, 0xc42026eba0, 0x4, 0x2, 0x0, 0x0, 0xc4202ddd18, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:146 +0x296
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7fb3d673e750, 0xc42026eba0, 0x1, 0x4, 0x0, 0x0, 0xc4202ddd18, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:88 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7fb3d673e750, 0xc42026eba0, 0xc4202ddd18, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:181 +0x76
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1()
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:162 +0xd6
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).runHeartbeat(0xc4203d7080, 0xc42032e000, 0xc42047b660, 0x14, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:322 +0x491
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).GRPCDial.func1.1.1()
        /go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:226 +0x69
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc42046cfc0, 0xc4203301b0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66
```

@mberhault what can I do to fix this panic?

**Comment by tamird** [comment]
Reviewed 13 of 13 files at r1, 2 of 5 files at r2.
Review status: 9 of 12 files reviewed at latest revision, 3 unresolved discussions.

---

*[pkg/acceptance/allocator_test.go, line 22 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744#-K_pXwj0Vo5GDVGGr-8f:-K_pXwj0Vo5GDVGGr-8g:byushhb) ([raw file](https://github.com/cockroachdb/cockroach/blob/c98a157b3dae7d7e5fdbe524e050bade24955e0d/pkg/acceptance/allocator_test.go#L22)):*
> ```Go
> // Required setup:
> // 1. Have an Azure account.
> // 2. Password-less SSH key in ~/.ssh/{azure,azure.pub}.
> ```

password here, passphrase below

---

*[pkg/acceptance/allocator_test.go, line 57 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744#-K_pXo0EH-4f7uwI9FQm:-K_pXo0EH-4f7uwI9FQn:b-huxfwf) ([raw file](https://github.com/cockroachdb/cockroach/blob/c98a157b3dae7d7e5fdbe524e050bade24955e0d/pkg/acceptance/allocator_test.go#L57)):*
> ```Go
> // - The minimum recommended version of Terraform is 0.7.2. If you see strange
> //   Terraform errors, upgrade your install of Terraform. Terraform 0.8.x or
> //   later might not work because of breaking changes to Terraform.
> ```

meaning 0.8.x wasn't tested, correct? would be nice to rephrase to avoid FUD

---

*[pkg/acceptance/terraform/common/download_binary.sh, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744#-K_pRVKxEbjOHzNijuhL:-K_pRVKxEbjOHzNijuhM:br3f600) ([raw file](https://github.com/cockroachdb/cockroach/blob/62a5865c75fc45a332e3c87728a91eb6a9834897/pkg/acceptance/terraform/common/download_binary.sh#L1)):*
> ```Shell
> #!/usr/bin/env bash
> ```

did you mean to add this file in the first commit?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744)*

**Comment by cuongdo** [comment]
*[pkg/acceptance/allocator_test.go, line 57 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744#-K_pXo0EH-4f7uwI9FQm:-K_pbKMFfi9Ah707HH45:b-7txfzb) ([raw file](https://github.com/cockroachdb/cockroach/blob/c98a157b3dae7d7e5fdbe524e050bade24955e0d/pkg/acceptance/allocator_test.go#L57)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

meaning 0.8.x wasn't tested, correct? would be nice to rephrase to avoid FUD
</blockquote></details>

There's already been one report of an issue with Terraform 0.8.0+, so it's certainly not FUD.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744)*

**Comment by cuongdo** [comment]
*[pkg/acceptance/terraform/common/download_binary.sh, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744#-K_pRVKxEbjOHzNijuhL:-K_pb5Cr_o8qEBkoV-HQ:b-efbrtf) ([raw file](https://github.com/cockroachdb/cockroach/blob/62a5865c75fc45a332e3c87728a91eb6a9834897/pkg/acceptance/terraform/common/download_binary.sh#L1)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

did you mean to add this file in the first commit?
</blockquote></details>

Yes. I wanted to pull it in from `cloud/gce`, as I don't like having a dependency on the `cloud` package.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744)*

**Comment by tamird** [comment]
Review status: 9 of 12 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/acceptance/allocator_test.go, line 57 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744#-K_pXo0EH-4f7uwI9FQm:-K_ptMaZ93RmF9p_E6aj:b-rf026l) ([raw file](https://github.com/cockroachdb/cockroach/blob/c98a157b3dae7d7e5fdbe524e050bade24955e0d/pkg/acceptance/allocator_test.go#L57)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

There's already been one report of an issue with Terraform 0.8.0+, so it's certainly not FUD.
</blockquote></details>

Hm, OK. In that case I'd prefer mentioning that issue here, but I won't insist.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12744)*

**Comment by cuongdo** [comment]
@mberhault any comments? it'd be nice to get our nightlies off of GCE, but I'm not sure how to progress with the clock issues

**Comment by bdarnell** [comment]
If you set COCKROACH_MAX_OFFSET=500ms, does it pass reliably?

**Comment by cuongdo** [comment]
I'll try the change to `COCKROACH_MAX_OFFSET` and see what happens.

**Comment by cuongdo** [comment]
5 of 5 runs passed with `COCKROACH_MAX_OFFSET=500ms`. Seems solid to me.

**Comment by mberhault** [comment]
LGTM

**Comment by cuongdo** [comment]
The continuous load test (runs `block_writer` against 4 nodes) sometimes fails when running on Azure, even with the max offset set to 500 ms. I'll try 1s just to see if that eliminates failures.

**Comment by cuongdo** [comment]
I've converted the public IP allocation from `static` to `dynamic`. Also, it appears that `COCKROACH_MAX_OFFSET=1s` is significantly more reliable for the 4-node `block_writer` load test.

@jordanlewis can you have a quick look at this? I know you had some misgivings about using a static storage account for VM VHDs, but I ran into issue after issue creating the storage account dynamically. There's some combination of Azure and Terraform bugs that make dynamically creating storage accounts really unreliable for this Terraform config.

File: pkg/acceptance/terraform/azure/variables.tf [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+# List of stores each with its own --store flag. (--store flags from the cockroach binary).
+variable "stores" {
+  default = "--store=data"
+}
+
+# Azure location to use.
+variable "azure_location" {
+  default = "eastus"
```
Comment (cuongdo) [inline pkg/acceptance/terraform/azure/variables.tf]: Let's try `eastus` first to reduce potential for SSH latency issues. This is simple enough to change if it becomes an issue.

**Pull request merged**